csharp/src/SeedLang/Interpreter/Chunk.cs

```diff
@@ -15,48 +15,68 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Text;
+using SeedLang.Common;
 
 namespace SeedLang.Interpreter {
   // A data structure to hold bytecode and constants generated from the AST tree by the compiler.
   internal class Chunk {
     // The maximum number of registers that can be allocated in the stack of a chunk.
     public const uint MaxRegisterCount = 250;
 
+    // The bytecode of this chunk.
+    public IReadOnlyList<Instruction> Bytecode => _bytecode;
+
+    // Source code ranges of the instructions in bytecode.
+    //
+    // The length of Bytecode and Range lists shall be the same.
+    public IReadOnlyList<Range> Ranges => _ranges;
+
     // The actual count of the registers that is needed for this chunk.
-    public uint RegisterCount = 0;
+    public uint RegisterCount { get; set; }
 
     private readonly List<Instruction> _bytecode = new List<Instruction>();
+
     // The constant list to hold all the constants used in this chunk.
-    private readonly List<Value> _constants = new List<Value>();
+    private readonly List<VMValue> _constants = new List<VMValue>();
+
+    private readonly List<Range> _ranges = new List<Range>();
 
     public override string ToString() {
       var sb = new StringBuilder();
-      foreach (var instr in _bytecode) {
-        sb.AppendLine($"{instr,-20}{ConstantOperandToString(instr)}");
+      for (int i = 0; i < _bytecode.Count; ++i) {
+        sb.Append($"{_bytecode[i],-20}{ConstantOperandToString(_bytecode[i])}");
+        sb.AppendLine(i < _ranges.Count && _ranges[i] is Range range ? $"{range}" : "");
```

Comment:
If the code logic should ensure that "The length of Bytecode and Range lists shall be the same", it'd be better to guard "I < _ranges.Count" with an Assert in debug mode, right? Does "_ranges[i] is Range" fall in the same situation?